pkg/pillar/devicenetwork/dhcpcd.go

```diff
@@ -211,16 +214,30 @@ func doDhcpClientInactivate(log *base.LogObject, nuc types.NetworkPortConfig) {
 		log.Functionf("doDhcpClientInactivate(%s) DT_NONE is a no-op\n",
 			nuc.IfName)
 	case types.DT_STATIC, types.DT_CLIENT:
-		extras := []string{}
-		if !dhcpcdCmd(log, "--release", extras, nuc.IfName, false) {
-			log.Errorf("doDhcpClientInactivate: release failed for %s\n",
-				nuc.IfName)
-		}
-		for dhcpcdExists(log, nuc.IfName) {
-			log.Warnf("dhcpcd %s still running", nuc.IfName)
+		var extras []string
+		// Wait for a bit then give up
+		waitCount := 0
+		failed := false
+		for {
+			if !dhcpcdCmd(log, "--release", extras, nuc.IfName, false) {
+				log.Errorf("doDhcpClientInactivate: release failed for %s\n",
+					nuc.IfName)
+			}
+			if !dhcpcdExists(log, nuc.IfName) {
+				break
+			}
+			waitCount++
+			if waitCount >= 3 {
+				log.Errorf("doDhcpClientInactivate: dhcpcd %s still running, give up", nuc.IfName)
+				failed = true
+				break
```

Comment:
There is a reason we wait forever here; under high long and low memory there is no guaratee that the --release (which results in a signal to the dhcpcd process I think) will result in that process being scheduled and cleaning up and exit in 3 seconds. So we shouldn't abandon the wait forever unless we have a very good reason.

Reply:
Well, inside of source I can see waiting for up to 10 seconds for process gone (leave the comment in source tree). I have no ideas/indication why it hung, but seems we need something like kill instead of release after several retries. Without it we have watchdog problem.